src/pages/Homepage/Homepage.tsx

```diff
@@ -36,22 +36,21 @@ export default function Homepage(): JSX.Element {
 	}, [socket])
 
 	useEffect(() => {
-		if (socket) {
-			setIsLoading(false);
-			if (profileStorage) {
-				setProfile(profileStorage);
-			} else {
-				setProfile({
-					username: '',
-					avatar: {
-						bodyColor: BodyColor.Yellow,
-						bodyType: BodyType.Pencil,
-						faceType: FaceType.Happy,
-						rubberColor: RubberColor.Pink,
-					},
-				})
-			}
+		if (!socket) return;
+		setIsLoading(false);
+		if (profileStorage) {
+			setProfile(profileStorage);
+			return;
 		}
+		setProfile({
+			username: '',
+			avatar: {
+				bodyColor: BodyColor.Yellow,
+				bodyType: BodyType.Pencil,
+				faceType: FaceType.Happy,
+				rubberColor: RubberColor.Pink,
+			},
+		})
```

Comment:
We should be using ProfileFactory.create(); here, that's much more convenient.

Reply:
I dont know, thats not my code 😆.